yastd/dbus_api.md

```diff
@@ -0,0 +1,169 @@
+
+## get the list of available languages
+
+Iface: o.o.YaST.Installer1.Language
+
+methods:
+
+  ToInstall(array(string)) -> void
+    Set list of languages to install
+    Example:
+      ToInstall(["cs_CZ", "de_DE"]) -> () # only lang codes from AvailableLanguages is supported
+
+Properties (all read only):
+
+  AvailableLanguages -> array(array(string, string, dict(string, variant)))
```

Comment:
yeap, struct makes sense here

Reply:
Another thing, currently ruby-dbus implements *struct* as an *Array* but I can imagine a mapping from a *Struct* (or any class, with some declaratory helpers).

Which reminds me that *OpenStruct* should pass for a D-Bus dictionary (feature wish)

Reply:
openstruct usage is basically discouraged. So I am not sure if it should be used.

About Struct usage it will be nice, but I think it is hard to get from dbus signature names of variables...and mapping from array to struct is quite easy `response = MyStruct.new(*dbus_call)`